File: binding/src/libqdrawutil.cpp
#include <QColor>
#include <QRect>
#include <QTileRules>
#include <qdrawutil.h>
#include "libqdrawutil.h"
#include "libqdrawutil.hxx"

QTileRules* QTileRules_new(const QTileRules* other) {
	 return new QTileRules(*other);
}

QTileRules* QTileRules_new2(QTileRules* other) {
	 return new QTileRules(*other);
}

QTileRules* QTileRules_new3(int horizontalRule, int verticalRule) {
	 return new QTileRules(static_cast<Qt::TileRule>(horizontalRule), static_cast<Qt::TileRule>(verticalRule));
}

QTileRules* QTileRules_new4() {
	 return new QTileRules();
}

QTileRules* QTileRules_new5(const QTileRules* param1) {
	 return new QTileRules(*param1);
}

QTileRules* QTileRules_new6(int rule) {
	 return new QTileRules(static_cast<Qt::TileRule>(rule));
}

void QTileRules_CopyAssign(QTileRules* self, QTileRules* other) {
    *self = *other;
}

void QTileRules_MoveAssign(QTileRules* self, QTileRules* other) {
    *self = std::move(*other);
}

int QTileRules_Horizontal(const QTileRules* self) {
	return self->horizontal;
}

void QTileRules_SetHorizontal(QTileRules* self, int horizontal) {
	self->horizontal;
}

int QTileRules_Vertical(const QTileRules* self) {
	return self->vertical;
}

void QTileRules_SetVertical(QTileRules* self, int vertical) {
	self->vertical;
}

void QTileRules_Delete(QTileRules* self) {
    delete self;
}

File: binding/src/libqdrawutil.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qtilerules.html
pub const qtilerules = struct {

    /// New constructs a new QTileRules object.
    pub fn New(other: ?*anyopaque) QtC.QTileRules {
        return qtc.QTileRules_new(@ptrCast(other));
    }


    /// New2 constructs a new QTileRules object.
    pub fn New2(other: ?*anyopaque) QtC.QTileRules {
        return qtc.QTileRules_new2(@ptrCast(other));
    }


    /// New3 constructs a new QTileRules object.
    pub fn New3(horizontalRule: i32, verticalRule: i32) QtC.QTileRules {
        return qtc.QTileRules_new3(@intCast(horizontalRule), @intCast(verticalRule));
    }


    /// New4 constructs a new QTileRules object.
    pub fn New4() QtC.QTileRules {
        return qtc.QTileRules_new4();
    }


    /// New5 constructs a new QTileRules object.
    pub fn New5(param1: ?*anyopaque) QtC.QTileRules {
        return qtc.QTileRules_new5(@ptrCast(param1));
    }


    /// New6 constructs a new QTileRules object.
    pub fn New6(rule: i32) QtC.QTileRules {
        return qtc.QTileRules_new6(@intCast(rule));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QTileRules_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QTileRules_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Horizontal(self: ?*anyopaque, ) i32 {
        return qtc.QTileRules_Horizontal(@ptrCast(self));
    }

    pub fn SetHorizontal(self: ?*anyopaque, horizontal: i32) void {
        qtc.QTileRules_SetHorizontal(@ptrCast(self), @intCast(horizontal));
    }

    pub fn Vertical(self: ?*anyopaque, ) i32 {
        return qtc.QTileRules_Vertical(@ptrCast(self));
    }

    pub fn SetVertical(self: ?*anyopaque, vertical: i32) void {
        qtc.QTileRules_SetVertical(@ptrCast(self), @intCast(vertical));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QTileRules_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/drawutil.html#types
pub const enums = struct {
    pub const DrawingHint = enum {
        pub const OpaqueTopLeft: i32 = 1;
        pub const OpaqueTop: i32 = 2;
        pub const OpaqueTopRight: i32 = 4;
        pub const OpaqueLeft: i32 = 8;
        pub const OpaqueCenter: i32 = 16;
        pub const OpaqueRight: i32 = 32;
        pub const OpaqueBottomLeft: i32 = 64;
        pub const OpaqueBottom: i32 = 128;
        pub const OpaqueBottomRight: i32 = 256;
        pub const OpaqueCorners: i32 = 325;
        pub const OpaqueEdges: i32 = 170;
        pub const OpaqueFrame: i32 = 495;
        pub const OpaqueAll: i32 = 511;
    };

};
